packages/workflow-core/src/lib/utils/definition-validator/extensions-validator.ts

```diff
@@ -0,0 +1,118 @@
+        pluginKind === 'kyb' ||
+        pluginKind === 'kyc-session' ||
+        pluginKind === 'email' ||
+        pluginKind === 'kyc'
+      ) {
+        validateApiPlugin(plugin as unknown as ISerializableHttpPluginParams, states)
+      }
+      if (pluginKind === 'webhook') {
+        validateWebhookPlugin(plugin as unknown as SerializableWebhookPluginParams)
+      }
+      if (pluginKind === 'iterative') {
+        validateIterativePlugin(plugin as unknown as SerializableIterativePluginParams, states)
+      }
+      if (pluginKind === 'child') {
+        validateChildPlugin(plugin as unknown as ISerializableChildPluginParams)
+      }
+    });
+  }
+}
+const validateApiPlugin = (
+  plugin: ISerializableHttpPluginParams,
+  states: StateMachine<any, any, any>['states']
+): void => {
+  validateTransformers(plugin.name, plugin.request.transform)
+  plugin.response.transform && validateTransformers(plugin.name, plugin.response.transform)
+
+  validatePluginStateAction(plugin.stateNames, states, plugin.successAction, plugin.errorAction)
+};
+const validateWebhookPlugin = (
+  plugin: SerializableWebhookPluginParams,
+): void => {
+  validateTransformers(plugin.name, plugin.request.transform)
+};
+const validateIterativePlugin = (
+  plugin: SerializableIterativePluginParams,
+  states: StateMachine<any, any, any>['states']
+): void => {
+  validateMapping('jmespath', plugin.iterateOn)
+  validatePluginStateAction(plugin.stateNames, states, plugin.successAction, plugin.errorAction)
+};
+const validateChildPlugin = (
+  plugin: ISerializableChildPluginParams
+): void => {
+  plugin.transformers?.forEach(transform => {
+    validateMapping(transform.transformer as Parameters<typeof validateMapping>[0], transform.mapping as Parameters<typeof validateMapping>[1])
+  })
+};
+
+const validateTransformers = (pluginName: string, transformers: SerializableValidatableTransformer['transform']) => {
+  transformers.forEach(transform => {
+    try {
+      transform.transformer === 'jmespath' &&
+      validateMapping('jmespath', transform.mapping as string);
+      transform.transformer === 'json-logic' &&
+      validateMapping('json-logic', transform.mapping as unknown as Record<string, unknown>)
+    } catch (ex) {
+      throw new Error(`Invalid transform for ${pluginName} - type: ${transform.transformer} transformer: ${isErrorWithMessage(ex) ? ex.message : ''}`);
+    }
+  });
+}
+
+const validatePluginStateAction = (
+  pluginStateNames: Array<string>,
+  states: Parameters<typeof extensionsValidator>[1],
+  successAction?: string,
+  errorAction?: string,
+) => {
+  if (pluginStateNames && successAction) {
+    pluginStateNames.forEach(stateName => {
+      validateCallbackTransition(states, stateName, successAction, 'successAction')
+    })
+  }
+
+  if (pluginStateNames && errorAction) {
+    pluginStateNames.forEach(stateName => {
+      validateCallbackTransition(states, stateName, errorAction, 'errorAction')
+    })
+  }
```

Comment:
Consider using optional chaining for `pluginStateNames`. i.e. `pluginStateNames?.forEach`. Would simplify the if condition.